sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingInsertsMetrics.java

```diff
@@ -156,6 +156,19 @@ public void updateSuccessfulAndFailedRows(int totalRows, int failedRows) {
       }
     }
 
+    /** Record rpc latency histogram metrics. */
+    private void recordRpcLatencyMetrics() {
+      Histogram latencyHistogram =
+          BigQuerySinkMetrics.createRPCLatencyHistogram(
+              BigQuerySinkMetrics.RpcMethod.STREAMING_INSERTS);
+      Object[] rpcLatencyObjs = rpcLatencies().toArray();
+      double[] rpcLatencies = new double[rpcLatencyObjs.length];
```

Comment:
nit: might be wrong but can write like
```
List<Double> rpcLatencies = rpcLatencies().stream().map(latencyObj -> ((Duration) rpcLatencyObjs[i]).toMillis()).collection(toList());
latencyHistogram.update(rpcLatencies);
```

Reply:
Looks like there's a handful of custom map functions to map to specific primitives. I used this sequence. 
```
double[] rpcLatencies =
  rpcLatencies().stream().mapToDouble(duration -> duration.toMillis()).toArray();
```